lib/entry-index.js

```diff
@@ -158,63 +158,47 @@ module.exports.lsStream = lsStream
-  const stream = from.obj()
-
-  // "/cachename/*"
-  readdirOrEmpty(indexDir)
-    .then((buckets) => {
-      return Promise.all(
-        buckets.map((bucket) => {
-          const bucketPath = path.join(indexDir, bucket)
-
-          // "/cachename/<bucket 0xFF>/*"
-          return readdirOrEmpty(bucketPath).then((subbuckets) => {
-            return Promise.all(
-              subbuckets.map((subbucket) => {
-                const subbucketPath = path.join(bucketPath, subbucket)
-
-                // "/cachename/<bucket 0xFF>/<bucket 0xFF>/*"
-                return readdirOrEmpty(subbucketPath).then((entries) => {
-                  return Promise.all(
-                    entries.map((entry) => {
-                      const getKeyToEntry = bucketEntries(
-                        path.join(subbucketPath, entry)
-                      ).then((entries) => {
-                        return entries.reduce((acc, entry) => {
-                          acc.set(entry.key, entry)
-                          return acc
-                        }, new Map())
-                      })
-
-                      return getKeyToEntry
-                        .then((reduced) => {
-                          for (const entry of reduced.values()) {
-                            const formatted = formatEntry(cache, entry)
-                            formatted && stream.push(formatted)
-                          }
-                        })
-                        .catch((err) => {
-                          if (err.code === 'ENOENT') {
-                            return undefined
-                          }
-                          throw err
-                        })
-                    })
-                  )
-                })
+  const stream = new Minipass({ objectMode: true })
+
+  readdirOrEmpty(indexDir).then(buckets => Promise.all(
+    buckets.map(bucket => {
+      const bucketPath = path.join(indexDir, bucket)
+      return readdirOrEmpty(bucketPath).then(subbuckets => Promise.all(
+        subbuckets.map(subbucket => {
+          const subbucketPath = path.join(bucketPath, subbucket)
+
+          // "/cachename/<bucket 0xFF>/<bucket 0xFF>./*"
+          return readdirOrEmpty(subbucketPath).then(entries => Promise.all(
+            entries.map(entry => {
+              const entryPath = path.join(subbucketPath, entry)
+              return bucketEntries(entryPath).then(entries =>
+                // using a Map here prevents duplicate keys from
+                // showing up twice, I guess?
+                entries.reduce((acc, entry) => {
+                  acc.set(entry.key, entry)
+                  return acc
+                }, new Map())
+              ).then(reduced => {
+                // reduced is a map of key => entry
+                for (const entry of reduced.values()) {
+                  const formatted = formatEntry(cache, entry)
+                  if (formatted) {
+                    stream.write(formatted)
```

Comment:
[question] why the change from `stream.push` to `stream.write` in this one?

Reply:
See above about Node core's 'Readable.push()` method.  Base Minipass streams don't have a "push()".  But the base Minipass class is a PassThrough stream, so `stream.write()` queues the data up to be emitted by a `'data'` event (with appropriate buffering, etc.)

Reply:
See above about `Minipass.write()`.  Same logical meaning, different stream implementation :)